Implement overwrite support for upload

arkal opened the pull request:
(resolves #30, resolves #46)

If a key already exists in the bucket, then exit with a message saying the file already exists. If the user wants to overwrite the
destination file, they can use the --overwrite flag.

hannes-ucsc commented on src/s3am/cli.py:
```diff
@@ -141,6 +142,9 @@ def add_common_arguments( sp ):
                           "exit with an error if it detects unfinished uploads." )
 
+    upload_sp.add_argument( '--overwrite', action='store_true',
+                            help='If the object already exists in S3, should it be overwritten?' )
```
"Overwrite the object at DST_URL if it already exists. Without this option, the existing object will not be changed and an the program exits with an error."

hannes-ucsc commented on src/s3am/operations.py:
```diff
@@ -339,6 +339,7 @@ def __init__( self, src_url, dst_url,
         :param str dst_url: URL of object in S3 to upload to
         :param bool resume: whether to resume an incomplete multipart upload
         :param bool force: whether to remove any incomplete multipart uploads before proceeding
+        :param bool overwrite: should objects that already exist in the bucket be overwritten
```
"overwrite existing objects or raise a UserError"

arkal committed:
Implement unique exit codes for various User Errors (resolves #46)

resolves #46

arkal committed:
Implement overwrite support for uploads (resolves #30)

resolves #30

Users can now choose to overwrite or skip uploads of files that already exist at the destination.

arkal commented:
@hannes-ucsc this is ready!

hannes-ucsc commented:
Continued in #48
